crates/payload/basic/src/lib.rs

```diff
@@ -316,6 +316,32 @@ pub enum PayloadState<P> {
     Frozen(P),
 }
 
+impl<P> PayloadState<P> {
+    /// Checks if the payload is frozen.
+    pub const fn is_frozen(&self) -> bool {
+        matches!(self, Self::Frozen(_))
+    }
+
+    /// Returns the payload if it exists (either Best or Frozen).
+    pub const fn payload(&self) -> Option<&P> {
+        match self {
+            Self::Missing => None,
+            Self::Best(p) | Self::Frozen(p) => Some(p),
+        }
+    }
+
+    /// Returns a clone of the payload if it exists (either Best or Frozen).
+    pub fn best_payload(&self) -> Option<P>
+    where
+        P: Clone,
+    {
+        match self {
+            Self::Missing => None,
+            Self::Best(p) | Self::Frozen(p) => Some(p.clone()),
+        }
+    }
```

Comment:
this fn is redundant because this is just `payload().cloned()`
so we can remove this